### PR TITLE
TSK-1161 Adapter must set field businessProcessId on taskana task.

### PR DESCRIPTION
**Pull request by BerndBreier**

### CODE REVIEW

**Comment by BerndBreier** [comment]
Sonar is green: https://sonarcloud.io/dashboard?branch=TSK-1161&id=BerndBreier_TaskanaAdapter

File: taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    assertThat(theTask.hashCode()).isNotNull();
+  }
+
+  @Test
+  void should_returnString_when_toStringIsCalled() {
+    assertThat(theTask.toString()).isNotNull();
+  }
+
+  @Test
+  void should_checkEgality_when_equalsIsCalled() {
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java:166]: typo, pls change checkEgality to checkEquality

File: taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    ReferencedTask refTask2 = theTask;
+    refTask2.setWorkbasketKey("nochnkey");
+    assertThat(refTask2.equals(theTask));
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java:169]: The assertion is not completed and therefore not asserting/testing. Please change this to assertThat(refTask2).equals(theTask);

File: taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    refTask2 = theTask;
+    assertThat(theTask.equals(refTask2));
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java:171]: please complete assertion, see above

File: taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    assertThat(!theTask.equals("aString"));
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java:172]: please complete assertion, see above. Also maybe you could use the isNotEqualTo assertion like this: assertThat(theTask).isNotEqualTo("aString)

File: taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+    assertThat(!theTask.equals(null));
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java:173]: please complete assertion and consider the isNotEqualTo assertion, see above

File: taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+    refTask2.setDomain("5");
+    refTask2.setDue("6");
+    refTask2.setId("7");
+    refTask2.setName("8");
+    refTask2.setOutboxEventId("9");
+    refTask2.setOutboxEventType("10");
+    refTask2.setOwner("11");
+    refTask2.setWorkbasketKey("anotherOne");
+
+    assertThat(!refTask2.equals(theTask));
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java:204]: please complete assertion and consider the isNotEqualTo assertion, see above

File: taskana-adapter-camunda-spring-boot-test/src/test/java/pro/taskana/adapter/integration/TestTaskAcquisition.java [hunk 7]
```diff
@@ -56,8 +56,11 @@ public void user_task_process_instance_started_in_camunda_via_rest_should_result
       List<TaskSummary> taskanaTasks =
           this.taskService.createTaskQuery().externalIdIn(camundaTaskId).list();
       assertEquals(1, taskanaTasks.size());
-      String taskanaTaskExternalId = taskanaTasks.get(0).getExternalId();
+      TaskSummary taskanaTaskSummary = taskanaTasks.get(0);
+      String taskanaTaskExternalId = taskanaTaskSummary.getExternalId();
       assertEquals(taskanaTaskExternalId, camundaTaskId);
+      String businessProcessId = taskanaTaskSummary.getBusinessProcessId();
+      assertEquals(processInstanceId, businessProcessId);
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-spring-boot-test/src/test/java/pro/taskana/adapter/integration/TestTaskAcquisition.java:63]: Pls change this to assertJ

File: taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+    assertThat(theTask.hashCode()).isNotNull();
+  }
+
+  @Test
+  void should_returnString_when_toStringIsCalled() {
+    assertThat(theTask.toString()).isNotNull();
+  }
+
+  @Test
+  void should_checkEgality_when_equalsIsCalled() {
```
Comment (gitgoodjhe) [inline taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java:166]: same typo

File: taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+    refTask2.setDomain("5");
+    refTask2.setDue("6");
+    refTask2.setId("7");
+    refTask2.setName("8");
+    refTask2.setOutboxEventId("9");
+    refTask2.setOutboxEventType("10");
+    refTask2.setOwner("11");
+    refTask2.setWorkbasketKey("anotherOne");
+
+    assertThat(!refTask2.equals(theTask));
```
Comment (gitgoodjhe) [inline taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java:204]: please complete assertion and consider the isNotEqualTo assertion, see above

File: taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+
+import org.junit.jupiter.api.Test;
+
+/** Acceptance test for ReferencedTask setters and getters. */
+public class ReferencedTaskTest {
+
+  ReferencedTask theTask;
+  String theValue;
+
+  public ReferencedTaskTest() {
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-listener/src/test/java/pro/taskana/adapter/camunda/dto/ReferencedTaskTest.java:13]: I'm not sure if we should forcefully test getter/setter, just to satisfy the code coverage in sonarcloud...maybe we should discuss this in daily/cop...i think last year we decided against it but i could remember this wrong

File: taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+    assertThat(theTask.toString()).isNotNull();
+  }
+
+  @Test
+  void should_checkEgality_when_equalsIsCalled() {
+    ReferencedTask refTask2 = theTask;
+    refTask2.setWorkbasketKey("nochnkey");
+    assertThat(refTask2.equals(theTask));
+    refTask2 = theTask;
+    assertThat(theTask.equals(refTask2));
```
Comment (gitgoodjhe) [inline taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java:171]: please complete assertion

File: taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+    assertThat(!theTask.equals("aString"));
```
Comment (gitgoodjhe) [inline taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java:172]: please complete assertion and consider the isNotEqualTo seertion, see above

File: taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+    assertThat(!theTask.equals(null));
```
Comment (gitgoodjhe) [inline taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java:173]: please complete assertion and consider the isNotEqualTo seertion, see above

File: taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+  @Test
+  void should_returnString_when_toStringIsCalled() {
+    assertThat(theTask.toString()).isNotNull();
+  }
+
+  @Test
+  void should_checkEgality_when_equalsIsCalled() {
+    ReferencedTask refTask2 = theTask;
+    refTask2.setWorkbasketKey("nochnkey");
+    assertThat(refTask2.equals(theTask));
```
Comment (gitgoodjhe) [inline taskana-adapter/src/test/java/pro/taskana/adapter/systemconnector/api/ReferencedTaskTest.java:169]: please complete assertion

**Pull request merged**